libs/template/class/src/lib/class.directive.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  Directive,
+  ElementRef,
+  Input,
+  IterableChanges,
+  IterableDiffer,
+  IterableDiffers,
+  KeyValueChanges,
+  KeyValueDiffer,
+  KeyValueDiffers,
+  OnDestroy,
+  OnInit,
+  Renderer2,
+  ɵisListLikeIterable as isListLikeIterable,
+} from '@angular/core';
+import {
+  accumulateObservables,
+  RxStrategyNames,
+  RxStrategyProvider,
+} from '@rx-angular/cdk';
+import { coerceAllFactory } from '@rx-angular/cdk/coercing';
+import {
+  BehaviorSubject,
+  isObservable,
+  Observable,
+  ObservableInput,
+  ReplaySubject,
+  Subscription,
+} from 'rxjs';
+import { map, switchAll, switchMap, withLatestFrom } from 'rxjs/operators';
+
+type NgClassValues =
+  | string[]
+  | Set<string>
+  | Record<string, boolean>
+  | string
+  | null
+  | undefined;
+
+type RxClassInput =
+  | Observable<NgClassValues>
+  | Record<string, Observable<boolean>>;
+
+@Directive({ selector: '[rxClass]' })
+export class ClassDirective implements OnInit, OnDestroy {
+  private iterableDiffer: IterableDiffer<string> | null = null;
+  private keyValueDiffer: KeyValueDiffer<string, any> | null = null;
+  private initialClasses: string[] = [];
+  private rawClass: NgClassValues = null;
+  private newInput = false;
+  private sub = new Subscription();
+
+  constructor(
+    private readonly iterableDiffers: IterableDiffers,
+    private readonly keyValueDiffers: KeyValueDiffers,
+    private readonly ngEl: ElementRef,
+    private readonly renderer: Renderer2,
+    private readonly strategyProvider: RxStrategyProvider
+  ) {}
+
+  private rxClass = coerceAllFactory<NgClassValues>(
+    () => new ReplaySubject(1),
+    switchAll()
+  );
+
+  @Input('rxClass')
+  set rxClass$(value$: RxClassInput) {
```

Comment:
Let's make it also accept static values. There is a PR that intros that already for let and push.
Should be merged already into main.

Reply:
That would be really cool. The issue here is, it has to wait for refactoring of `accumulateObservables` which doesn't accept static values at the moment and is used in the current `RxClass` implementation.

![image](https://user-images.githubusercontent.com/22128812/139813745-c06e9114-179b-471b-8a19-21e32375eb3c.png)

Reply:
no, we need to create an issue and work on it.